Allow plot-{x,y,z}-far-tick-labels? to have three options (issue #46)

alex-hhh opened the pull request:
The plot-{x,y,z}-far-tick-labels? parameters can have three possible values:
'auto, #t or #f.

When a parameter is set to 'auto, the behavior is that far labels are present
only if the ticks on the far axis are different from the ticks on the near
axis (this was the plot behavior before this parameter was introduced).

When a parameter is set to #t or #f, the far labels are shown or not
respectively.  Note that labels will only be displayed if the axis ticks are
drawn on the far axis.

bennn commented on plot-lib/plot/private/plot2d/plot-area.rkt:
```diff
@@ -337,8 +337,16 @@
+      (let ((flag (plot-x-far-tick-labels?)))
+        (if (eq? flag 'auto)
+            (not (and (plot-x-axis?) (equal? x-ticks x-far-ticks)))
+            flag)))
```
Maybe replace this code with a function call `(f (plot-x-far-tick-labels?) (plot-x-axis?) x-ticks x-far-ticks)` ?

I can't think of what to name `f` though, so maybe not.

bennn commented on plot-test/plot/tests/pr46-test.rkt:
```diff
@@ -0,0 +1,60 @@
+               [plot-y-far-tick-labels? #t]
+               [plot-y-ticks (linear-ticks)]
+               [plot-y-far-ticks (linear-ticks)])
+  (plot (function sin (- pi) pi)))
```
Can you set the x-tick parameters (the same way as the y-tick params) in all these tests, and add a few 3d tests for the z-params

alex-hhh commented:
I have updated the examples to include 3d plots -- it looks like the default `'auto` behavior for 3d plots is different than for the 2D plots, so I just changed the text to print out the parameter values that are used in each case.

With regards to documentation, I am not sure that the 'auto parameter should be documented -- it is the default when the user does not explicitly ask for far axis labels, I think the user should only ever set this parameter to `#t` or `#f`.  What do you think?

bennn commented:
> With regards to documentation, I am not sure that the 'auto parameter should be documented -- it is the default when the user does not explicitly ask for far axis labels, I think the user should only ever set this parameter to #t or #f. What do you think?

Sounds good. Can you delete the `#:value #f` in the docs, as in:

```
(defparam plot-x-far-tick-labels? draw? boolean? #:value #f) 
```

bennn commented on plot-test/plot/tests/pr46-test.rkt:
```diff
@@ -0,0 +1,134 @@
+#lang racket
```
Oh, can you move this file to `plot-test/plot/tests/PRs/46.rkt`

bennn commented:
thank you! https://github.com/racket/plot/commit/9823d1ab9ff9144a8a3114afc0af6a69a0fbc4b7
